invesalius/gui/preferences.py

```diff
@@ -207,6 +533,231 @@ def LoadSelection(self, values):
         self.rb_inter.SetSelection(int(surface_interpolation))
         self.rb_inter_sl.SetSelection(int(slice_interpolation))
 
+    def OnSelectColormap(self, event=None):
+        self.conf["mep_colormap"] = self.colormaps[self.combo_thresh.GetSelection()]
+        colors = self.GenerateColormapColors(self.conf.get("mep_colormap"), self.number_colors)
+
+        # Save the configuration
+        self.session.SetConfig("mep_configuration", self.conf)
+        Publisher.sendMessage("Save Preferences")
+        self.UpdateGradient(self.gradient, colors)
+
+    def GenerateColormapColors(self, colormap_name, number_colors=4):
+        # Extract colors and positions
+        color_def = const.MEP_COLORMAP_DEFINITIONS[colormap_name]
+        colors = list(color_def.values())
+        positions = [0.0, 0.25, 0.5, 1.0]  # Assuming even spacing between colors
+
+        # Create LinearSegmentedColormap
+        cmap = mcolors.LinearSegmentedColormap.from_list(
+            colormap_name, list(zip(positions, colors))
+        )
+        colors_gradient = [
+            (
+                int(255 * cmap(i)[0]),
+                int(255 * cmap(i)[1]),
+                int(255 * cmap(i)[2]),
+                int(255 * cmap(i)[3]),
+            )
+            for i in np.linspace(0, 1, number_colors)
+        ]
+
+        return colors_gradient
+
+    def UpdateGradient(self, gradient, colors):
+        gradient.SetGradientColours(colors)
+        gradient.Refresh()
+        gradient.Update()
+
+        self.Refresh()
+        self.Update()
+        self.Show(True)
+
+    def OnComboName(self, evt):
+        from invesalius import project as prj
+
+        self.proj = prj.Project()
+        surface_index = self.combo_brain_surface_name.GetSelection()
+        Publisher.sendMessage("Show only surface", surface_index=surface_index)
+        Publisher.sendMessage(
+            "Set MEP brain surface",
+            surface=self.proj.surface_dict[surface_index],
+            surface_index=surface_index,
+        )
+        Publisher.sendMessage("Get visible surface actor")
+
+        self.button_colour.SetColour(
+            [int(value * 255) for value in self.proj.surface_dict[surface_index].colour]
+        )
+
+    def OnSelectColour(self, evt):
+        colour = [value / 255.0 for value in self.button_colour.GetColour()]
+        Publisher.sendMessage(
+            "Set surface colour",
+            surface_index=self.combo_brain_surface_name.GetSelection(),
+            colour=colour,
+        )
+
+
+class LoggingTab(wx.Panel):
```

Comment:
This should not appers in this PR

Reply:
Ok, I removed this but It caused issues which made me remove some logging related logic in the frames.py file.. Is that ok?

Reply:
Sorry, the log stuff is in the invesalius master and should not be changed or removed.